crates/revm/src/witness.rs

```diff
@@ -22,7 +22,11 @@ pub struct ExecutionWitnessRecord {
 
 impl ExecutionWitnessRecord {
     /// Records the state after execution.
-    pub fn record_executed_state<DB>(&mut self, statedb: &State<DB>) {
+    pub fn record_executed_state<DB>(
+        &mut self,
+        statedb: &State<DB>,
+        #[cfg(feature = "scroll")] context: &reth_scroll_primitives::ScrollPostExecutionContext,
```

Comment:
this is required because of a cyclic dep: `reth-scroll-execution` needs to import `reth-revm`, to implement `State<StateProviderDatabase<DB>>`, and the trait can't be implemented elsewhere because of the orphan rule. This means `reth-revm` can't import `reth-scroll-execution`, so we can't have a bound on `State<DB>`.